sdk/metric/meter_test.go

```diff
@@ -2064,3 +2068,208 @@ func TestHistogramBucketPrecedenceOrdering(t *testing.T) {
+				"observable.int64.counter",
+				"observable.int64.up.down.counter",
+				"observable.int64.gauge",
+				"observable.float64.counter",
+				"observable.float64.up.down.counter",
+				"observable.float64.gauge",
+			},
+			expectedLogs: []map[string]interface{}{},
+		},
+	}
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			logEntries := []map[string]interface{}{}
+			global.SetLogger(
+				funcr.NewJSON(
+					func(obj string) {
+						var entry map[string]interface{}
+						_ = json.Unmarshal([]byte(obj), &entry)
+						logEntries = append(logEntries, entry)
+					},
+					funcr.Options{Verbosity: 0},
+				),
+			)
+
+			reader := NewManualReader()
+			meter := NewMeterProvider(WithView(tt.views...), WithReader(reader)).Meter("TestObservableDropAggregation")
+
+			aiCounter, err := meter.Int64ObservableCounter("observable.int64.counter")
+			require.NoError(t, err)
+			aiUpDownCounter, err := meter.Int64ObservableUpDownCounter("observable.int64.up.down.counter")
+			require.NoError(t, err)
+			aiGauge, err := meter.Int64ObservableGauge("observable.int64.gauge")
+			require.NoError(t, err)
+
+			afCounter, err := meter.Float64ObservableCounter("observable.float64.counter")
+			require.NoError(t, err)
+			afUpDownCounter, err := meter.Float64ObservableUpDownCounter("observable.float64.up.down.counter")
+			require.NoError(t, err)
+			afGauge, err := meter.Float64ObservableGauge("observable.float64.gauge")
+			require.NoError(t, err)
+
+			unregisterediCounter, err := meter.Int64ObservableCounter("unregistered.observable.int64.counter")
+			require.NoError(t, err)
+			unregisteredfCounter, err := meter.Float64ObservableCounter("unregistered.observable.float64.counter")
+			require.NoError(t, err)
+
+			_, err = meter.RegisterCallback(
+				func(ctx context.Context, obs metric.Observer) error {
+					obs.ObserveInt64(aiCounter, 1)
+					obs.ObserveInt64(aiUpDownCounter, 1)
+					obs.ObserveInt64(aiGauge, 1)
+					obs.ObserveFloat64(afCounter, 1)
+					obs.ObserveFloat64(afUpDownCounter, 1)
+					obs.ObserveFloat64(afGauge, 1)
+
+					obs.ObserveInt64(unregisterediCounter, 1)
+					obs.ObserveFloat64(unregisteredfCounter, 1)
+
+					return nil
+				},
+				aiCounter, aiUpDownCounter, aiGauge, afCounter, afUpDownCounter, afGauge,
+			)
+			require.NoError(t, err)
+
+			var rm metricdata.ResourceMetrics
+			err = reader.Collect(context.Background(), &rm)
+			require.NoError(t, err)
+
+			if len(tt.expectedObservableName) == 0 {
+				require.Len(t, rm.ScopeMetrics, 0)
+				return
+			}
+
+			require.Len(t, rm.ScopeMetrics, 1)
+			require.Len(t, rm.ScopeMetrics[0].Metrics, len(tt.expectedObservableName))
+
+			for i, m := range rm.ScopeMetrics[0].Metrics {
+				assert.Equal(t, tt.expectedObservableName[i], m.Name)
+			}
+
+			if diff := cmp.Diff(logEntries, tt.expectedLogs,
+				cmpopts.IgnoreMapEntries(func(_ string, v interface{}) bool {
+					return v.(string) == "" // skip comparing empty values
```

Comment:
This looks wrong. I guess that the function inside `global.SetLogger(` invocation should make sure to not add such entries.

Then we should also be able to use `assert.Equal` instead of `cmp.Diff`.